ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -585,6 +594,17 @@ public func first<T, E>(producer: SignalProducer<T, E>) -> Result<T, E>? {
 	return result
 }
 
+/// Starts the producer, then blocks, waiting for the last value.
+public func last<T, E>(producer: SignalProducer<T, E>) -> Result<T, E>? {
+	return producer |> takeLast(1) |> single
+}
+
+/// Starts the producer, then blocks, waiting for completion.
+public func wait<T, E>(producer: SignalProducer<T, E>) -> Result<(), E> {
+	let result = producer |> map { _ in () } |> last
```

Comment:
I wouldn't be able to use `concat`, but could use `then`. If the `??` is still needed, is swapping `map` for `then` much of an improvement?